src/main/java/com/gradle/develocity/bamboo/ShortLivedTokenClient.java

```diff
@@ -0,0 +1,64 @@
+package com.gradle.develocity.bamboo;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class ShortLivedTokenClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShortLivedTokenClient.class);
+
+    private static final RequestBody EMPTY_BODY = RequestBody.create(new byte[]{});
+
+    private final OkHttpClient httpClient;
+
+    public ShortLivedTokenClient() {
+        this.httpClient = new OkHttpClient().newBuilder()
+                .connectTimeout(10, TimeUnit.SECONDS)
+                .writeTimeout(10, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .build();
+    }
+
+    public Optional<DevelocityAccessKey> get(String server, DevelocityAccessKey accessKey, String expiryInHours) {
```

Comment:
I don't think the comment is necessary as the method lives in the `ShortLivedTokenClient` which already explains this.
As for not introducing a separate class for short-lived token, we didn't do it for 2 reasons:
1. Implementation-wise, they'd be exactly the same as their format and purpose are identical. We might share the code via superclass, but still, we'd end up having two exact classes in the end.
2. Conceptually, short-lived access token is a form of an access key. For build agents, we even set it in the same environment variable (GRADLE_ENTEPRISE_ACCESS_KEY/DEVELOCITY_ACCESS_KEY). So one could argue there is some inconsistency there. So from the domain perspective, the concepts are already mixed a bit.

Therefore, we simply chose a pragmatic approach and stayed with one object. 
In case we need to distinguish them on some deeper level, we'd certainly introduce a separate class.

Reply:
Sounds good. I think the single class seems fine, and you make a good point about the "ShortLivedTokenClient" 👍